src/goto-instrument/synthesizer/loop_invariant_synthesizer.h

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************\
+
+Module: Loop Invariant Synthesizer Interface
+
+Author: Qinheping Hu
+
+\*******************************************************************/
+
+/// \file
+/// Loop Invariant Synthesizer Interface
+
+#ifndef CPROVER_GOTO_INSTRUMENT_SYNTHESIZER_LOOP_INVARIANT_SYNTHESIZER_H
+#define CPROVER_GOTO_INSTRUMENT_SYNTHESIZER_LOOP_INVARIANT_SYNTHESIZER_H
+
+#include <util/message.h>
+
+#include <goto-programs/goto_model.h>
+
+#include "synthesizer_utils.h"
+
+#define OPT_SYNTHESIZE_LOOP_INVARIANTS "(synthesize-loop-invariants)"
+#define HELP_LOOP_INVARIANT_SYNTHESIZER                                        \
+  " --synthesize-loop-invariants\n"                                            \
+  "                              synthesize and apply loop invariants\n"
+
+class loop_invariant_synthesizer_baset
+{
+public:
+  loop_invariant_synthesizer_baset(goto_modelt &goto_model, messaget &log)
```

Comment:
I would expect this to be `const goto_modelt &`

Reply:
We apply loop contracts in the ```goto_model``` in the implementation of the synthesizer, which is contained in the later PR. So we need to keep it non const.